console/console-init/ui/src/modules/iot-device/components/MangeColumnModal/ManageColumnModal.tsx

```diff
@@ -0,0 +1,369 @@
+      case "lastUpdated":
+        setCheckedColumns({ ...checkedColumns, isLastUpdatedChecked: checked });
+        break;
+      case "lastSeen":
+        setCheckedColumns({ ...checkedColumns, isLastSeenChecked: checked });
+        break;
+      case "addedDate":
+        setCheckedColumns({ ...checkedColumns, isAddedDateChecked: checked });
+        break;
+      case "memberOf":
+        setCheckedColumns({ ...checkedColumns, isMemberOfChecked: checked });
+        break;
+      case "viaGateways":
+        setCheckedColumns({ ...checkedColumns, isViaGatewaysChecked: checked });
+        break;
+    }
+  };
+  const onSave = () => {
+    const columns = [];
+    const {
+      isDeviceIdChecked,
+      isConnectionTypeChecked,
+      isStatusChecked,
+      isLastUpdatedChecked,
+      isLastSeenChecked,
+      isAddedDateChecked,
+      isMemberOfChecked,
+      isViaGatewaysChecked
+    } = checkedColumns;
+
+    if (isDeviceIdChecked) {
+      columns.push("deviceId");
+    }
+    if (isConnectionTypeChecked) {
+      columns.push("connectionType");
+    }
+    if (isStatusChecked) {
+      columns.push("status");
+    }
+    if (isLastUpdatedChecked) {
+      columns.push("lastUpdated");
+    }
+    if (isLastSeenChecked) {
+      columns.push("lastSeen");
+    }
+    if (isAddedDateChecked) {
+      columns.push("addedDate");
+    }
+    if (isMemberOfChecked) {
+      columns.push("memberOf");
+    }
+    if (isViaGatewaysChecked) {
+      columns.push("viaGateways");
+    }
+    setSelectedColumns(columns);
+    handleModalToggle();
+  };
+
+  const {
+    isDeviceIdChecked,
+    isConnectionTypeChecked,
+    isStatusChecked,
+    isLastUpdatedChecked,
+    isLastSeenChecked,
+    isAddedDateChecked,
+    isMemberOfChecked,
+    isViaGatewaysChecked
+  } = checkedColumns;
+
+  return (
+    <>
+      <Modal
+        title="Manage columns"
+        isOpen={isModalOpen}
+        id="device-list-manage-column-modal"
+        variant="small"
+        description={
+          <TextContent>
+            <Text component={TextVariants.p}>
+              Checked categories will be displayed in the table.
+            </Text>
+          </TextContent>
+        }
+        onClose={handleModalToggle}
+        actions={[
+          <Button
+            key="save"
+            id="manage-column-save-button"
+            variant="primary"
+            onClick={onSave}
+          >
+            Save
+          </Button>,
+          <Button
+            key="cancel"
+            id="manage-column-cancel-button"
+            variant="secondary"
+            onClick={handleModalToggle}
+          >
+            Cancel
+          </Button>
+        ]}
+      >
+        <DataList
+          aria-label="Table column management"
+          id="device-list-column-management"
+          isCompact
+        >
+          <DataListItem
+            id="device-list-col-mangmnt-deviceid"
+            aria-labelledby="table-column-management-device-id"
```

Comment:
don't need to add dash "-" with aria-labelledby or any aria tags. it should be simple text.